tuna.go

```diff
@@ -498,6 +580,21 @@ func (c *Common) wrapConn(conn net.Conn, remotePublicKey []byte, localConnMetada
 	}
 
 	encryptKey := computeEncryptKey(connNonce, sharedKey[:])
+	k := string(append(remotePublicKey, connNonce...))
+	c.encryptKeys.Store(k, encryptKey)
+
+	if c.IsServer {
+		c.udpReadyChanLock.Lock()
+		_, ok := c.connReadyChan[k]
+		if !ok {
+			c.connReadyChan[k] = make(chan struct{}, 1)
+		}
+		select {
+		case c.connReadyChan[k] <- struct{}{}:
```

Comment:
You shouldn't put chan read/write inside lock, will cause deadlock

Reply:
Now map read is NOT protected by lock. You need to make sure:
1. Map read/write are ALWAYS protected by lock
2. Related (lock protected) channel read/write are not inside lock